great_expectations/self_check/util.py

```diff
@@ -1516,16 +1516,27 @@ def generate_expectation_tests(
     for d in test_data_cases:
         d = copy.deepcopy(d)
         dialects_to_include = {}
+        engines_to_include = {}
 
         # Some Expectations (mostly contrib) explicitly list test_backends/dialects to test with
         if d.test_backends:
             for tb in d.test_backends:
+                engines_to_include[tb["backend"]] = True
```

Comment:
Can `ExpectationTestDataCases` be improved here with additional types and typing? Especially for `test_backends` where there are clearly expected keys of `"backend"` and `"dialects"`

Reply:
Do you have any suggestions for implementing?

Reply:
yes, introduce a type called `TestBackend` that includes two fields, `backend` and `dialect`, then have the `test_backends` property of `ExpectationTestDataCases` return an `Optional[List[TestBackend]]`